src/backend/PortabilityService.Functions/Report.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Host;
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+
+namespace PortabilityService.Functions
+{
+    public static class Report
+    {
+        [FunctionName("report")]
+        public static HttpResponseMessage Run(
+            [HttpTrigger(AuthorizationLevel.Function, "get", Route = "report/{submissionId}")] HttpRequestMessage req,
+            string submissionId,
+            TraceWriter log)
+        {
+            if (!ValidAccessKey(req))
+            {
+                return req.CreateResponse(HttpStatusCode.Unauthorized);
+            }
+
+            // simulate report generation taking some time
+            if (new Random().Next(10) < 4)
+            {
+                return req.CreateResponse(HttpStatusCode.NotFound);
+            }
+
+            switch (req.Headers.Accept.ToString())
+            {
+                case "application/json":
+                    return JsonReport(req);
+                case "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet":
+                    return ExcelReport(req);
+                default:
+                    return req.CreateResponse(HttpStatusCode.UnsupportedMediaType);
+            }
+        }
+
+        public static bool ValidAccessKey(HttpRequestMessage request)
```

Comment:
As long as this is temporary, can we put a TODO to replace with authorization or investigate?  I think we have a user story for it.. not sure.